kartothek/io_components/metapartition.py

```diff
@@ -1549,6 +1549,67 @@ def delete_from_store(self, dataset_uuid, store):
             store.delete(file_key)
         return self.copy(files={}, data={}, metadata={})
 
+    def get_parquet_metadata(self, store, table_name):
+        """
+        Retrieve the parquet metadata for the MetaPartition.
+        Especially relevant for calculating dataset statistics.
+
+        Parameters
+        ----------
+        store
+          A factory function providing a KeyValueStore
+        table_name
+          Name of the kartothek table for which the statistics should be retrieved
+
+        Returns
+        -------
+        pd.DataFrame
+          A DataFrame with relevant parquet metadata
+        """
+        if not isinstance(table_name, str):
+            raise TypeError("Expecting a string for parameter `table_name`.")
+
+        if callable(store):
+            store = store()
+
+        if self.files:
+            fd = store.open(self.files[table_name])
+            pq_metadata = pa.parquet.ParquetFile(fd).metadata
+            data = {
+                "partition_label": self.label,
+                "number_rows_total": pq_metadata.num_rows,
+                "number_row_groups": pq_metadata.num_row_groups,
+                "serialized_size": pq_metadata.serialized_size,
+            }
+            data["row_group_id"] = range(data["number_row_groups"])
+
+            data["row_group_byte_size"], data["number_rows_per_row_group"] = zip(
+                *[
+                    (
+                        pq_metadata.row_group(rg_id).total_byte_size,
+                        pq_metadata.row_group(rg_id).num_rows,
+                    )
+                    for rg_id in data["row_group_id"]
+                ]
+            )
+            return pd.DataFrame(
+                data=data,
+                columns=[
+                    "partition_label",
+                    "row_group_id",
+                    "row_group_byte_size",
+                    "number_rows_total",
+                    "number_row_groups",
+                    "serialized_size",
+                    "number_rows_per_row_group",
+                ],
+            )
+
+        else:
+            raise RuntimeError(
+                "This should not happen: MetaPartition with no data associated."
```

Comment:
It can: if the metapartition was created from data (DF) and was not preserved/stored yet.

Reply:
Theoretically yes. In this particular workflow not and in any case, this exception should be fine. Maybe change the wording / remove the "this should not happen" thing. At least for the workflows where this is / should be used in this _should_ not happen

Reply:
👍 for changing the wording to something like "MetaPartition.get_parquet_metadata requires the MetaPartition to have files. If you have in-memory data, you must store it first".

Reply:
This error was removed as the logic was changed to take into account partitions that don't have files for the table queried